pkg/signature/options.go

```diff
@@ -55,3 +56,50 @@ type VerifyOption interface {
 	RPCOption
 	MessageOption
 }
+
+type signerVerifierOpts struct {
+	hashFunc      crypto.Hash
+	useED25519ph  bool
+	rsaPSSOptions *rsa.PSSOptions
```

Comment:
`PSSOptions` does include a hash function as an option. It is well-defined if both are set - https://pkg.go.dev/crypto/rsa#PSSOptions notes that what's in `PSSOptions` will override the hash function passed to it. Not a significant issue, but just something to note.

Reply:
Shall we check that the Hash in PSSOptions is the same as the hash specified in `signerVerifierOpts`? We could do that check inside `makeSignerVerifierOpts`.

Reply:
I'm fine with PSSOptions overriding hashFunc, maybe add a comment about this?